run.go

```diff
@@ -326,38 +329,96 @@ func Log(c chan *SimulationStatus) DomainManipulator {
 	}
 }
 
-func (d *InMAP) checkOutputNames(n ...string) error {
+func removeDuplicates(s []string) []string {
+	result := []string{}
```

Comment:
To minimize the number of memory allocations required in this function, instead do 'result := make([]string, 0, len(s)'. This initializes an array with zero length but will enough room available to grow it to the size of s without having to reallocate memory.